File: src/09.tree/31.404.go
package tree

func SumOfLeftLeaves(root *TreeNode) int {
	var dfs func(node *TreeNode, isLeft bool)
	ans := 0
	dfs = func(node *TreeNode, isLeft bool) {
		if node == nil {
			return
		}

		if isLeft && node.Left == nil && node.Right == nil {
			ans += node.Val
		}

		dfs(node.Left, true)
		dfs(node.Right, false)
	}

	dfs(root, false)

	return ans
}

File: src/09.tree/31.404.ts
import { TreeNode } from "./01.100";

export function sumOfLeftLeaves(root: TreeNode | null): number {
    let ans = 0;
    const dfs = (node: TreeNode | null, isLeft: boolean) => {
        if (!node) {
            return;
        }

        if (isLeft && !node.left && !node.right) {
            ans += node.val;
        }

        dfs(node.left, true);
        dfs(node.right, false);
    };
    dfs(root, false);
    return ans;
}
